haskell.nix/flake.nix

```diff
@@ -0,0 +1,116 @@
+{
+  inputs = {
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+    nixpkgs.url = "github:serokell/nixpkgs";
+    haskell-nix.url = "github:serokell/haskell.nix/serokell-latest";
+    hackage-nix = {
+      url = "github:input-output-hk/hackage.nix";
+      flake = false;
+    };
+    stackage-nix = {
+      url = "github:input-output-hk/stackage.nix";
+      flake = false;
+    };
+
+    # if you want to run weeder:
+    ##haskell-nix-weeder = {
+    ##  url = "github:serokell/haskell-nix-weeder";
+    ##  flake = false;
+    ##};
+  };
+
+  outputs = { self, nixpkgs, haskell-nix, hackage-nix, stackage-nix, ... }:
+
+  # for weeder:
+  ##outputs = { self, nixpkgs, haskell-nix, hackage-nix, stackage-nix, haskell-nix-weeder, ... }:
+
+    let
+      haskellNix = import haskell-nix {
+        sourcesOverride = { hackage = hackage-nix; stackage = stackage-nix; };
+      };
+
+      pkgs = import nixpkgs haskellNix.nixpkgsArgs;
+      lib = pkgs.lib;
+
+      # invoke haskell.nix
+      hs-pkgs = pkgs.haskell-nix.stackProject {
+        src = pkgs.haskell-nix.haskellLib.cleanGit {
+          name = "pataq-package";
+          src = ./.;
+        };
+
+        # haskell.nix configuration
+        modules = [{
+          packages.pataq-package = {
+            ghcOptions = [
+              # fail on warnings
+              "-Werror"
+              # disable optimisations to speed up builds
+              "-O0"
```

Comment:
`-O0` is good for builds within PRs, but if we build something that will be launched (e. g. deployed to our server) or uploaded somewhere to potentially be used by somebody (e. g. to releases), `-O1` or `-O2` is likely desired.

We can have a flag or something to control optimization settings. Probably it shouldn't be part of this template to keep it simple, but I'd add a comment as a reminder for people who'll need to build something with optimizations.

Reply:
Yeah, there would be concerns in this case, e.g. how to producing a static executable/a docker image, or setting up deployment step, and the setup may depend on the project itself, so it's hard to include this into the template. But I tried to rephrase the comment now, to make it clearer that it's only for non-release builds.